httpapi/controller.go

```diff
@@ -0,0 +1,40 @@
+package httpapi
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"github.com/nibble-4bits/ondemand-go-bootcamp/usecase"
+)
+
+func getPokemonByIDController(service usecase.PokemonService) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		idStr := c.Param("id")
+		id, err := strconv.Atoi(idStr)
+		if err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+			return
+		}
+
+		pokemon, err := service.GetByID(id)
+		if err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
```

Comment:
Actually I was thinking of something more along the lines of `http.StatusNotFound`, since the server logic worked properly but the actual error was that the pokemon corresponding to that ID wasn't found (because it doesn't exist).